02-documentation/developer-guides/building-a-nextjs-blog-with-squidex.md

```diff
@@ -0,0 +1,585 @@
+
+Now, to get the token, click the **Connect** button for the `nextjs-blog` client. From the list of connection methods provided, select **Connect manually**. A token will be generated and will be displayed below the text reading **Just use the following token**. Click the copy icon to copy the generated token.
+
+After copying this token with Read access to your application, you will create a file containing environment variables for your Next.js app. Create a `.env.local` file in the `squidex-blog` folder and add the following contents
+
+```
+SQUIDEX_API_URL=https://cloud.squidex.io/api/content/squidex-blog/graphql
+SQUIDEX_BEARER_TOKEN=API-KEY
+```
+
+Replace `squidex-blog` in SQUIDEX_API_URL with the name of your app on Squidex and replace `API-KEY` with the token you copied from the dashboard.
+
+These environment variables will not be visible on the client-side bundle since they will only be used at build time to authenticate queries to the Squidex API. In Next.js, the only environment variables that are available on client-side are those prefixed with `NEXT_PUBLIC_`. You can learn more about environment variables in Next.js [here](https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser).
+
+**NOTE:** When you deploy your app to production, make sure to replace the Squidex token every month since the tokens are valid for only a month.
+
+Restart the development server to load the environment variables from `.env.local`. Press the combination `Ctrl/CMD + C` on your terminal to stop the development server, then restart it by running:
+
+```bash
+yarn dev
+```
+
+You will see a message on the terminal showing that environment variables have been loaded from `.env.local`.
+
+Now you have set up the environment variables for your Next.js blog, you will create a helper function that adds the authorization token to the Authorization header of GraphQL queries you will make to Squidex.
+
+Create a `lib` folder in  `squidex-blog` and in this `lib` folder, create a `squidex.js` file. This file will contain the `fetchAPI` helper function for fetching data from Squidex.
+
+Add the following contents to the `squidex-blog/lib/squidex.js` file:
+
+```js
+export default async function fetchAPI(query, { variables } = {}) {
```

Comment:
Do you have the source code somewhere?

Reply:
The article is written so that the reader creates everything from scratch. There is no source code that the reader works based on.

Reply:
Yes, I know, but it would be valuable.

Reply:
I created a repo with the completed code at https://github.com/okezieuc/nextjs-squidex-blog.